### PR TITLE
trim prefix

### PR DESCRIPTION
**Pull request by tty2**
trim prefix method:
- cuts `n` cells from the beginning of the string
- adds `prefix` to string if set

### CODE REVIEW

**Comment by mattn** [comment]
I don't understand why it's `xxx...` instead of `...xxx`

**Comment by tty2** [comment]
Cause it's a tail.
To be honest I'm not really sure if tail is necessary in this function. I just wanted to keep similarity with `Truncate`. But in my case it's not necessary.
Do you mean that it's more obvious to set prefix? It's easy to implement as well.

**Comment by tty2** [comment]
Yes. I took a look at it more careful. You are right. It's stupid of me. Let me refactor this.

**Comment by mattn** [comment]
Thanks. I'll look this in later.

File: runewidth_test.go [hunk 1]
```diff
@@ -380,6 +380,76 @@ func TestTruncateNoNeeded(t *testing.T) {
 	}
 }
 
+func Test_TrimPrefix(t *testing.T) {
```
Comment (mattn) [inline runewidth_test.go:383]: Remove `_`

**Comment by mattn** [comment]
Thanks your contribution. I prefer the tests should be table-driven tests. See other tests.

**Comment by tty2** [comment]
Sure. I should have written them similar.
Changed.

**Comment by mattn** [comment]
Hmm, I'm still wondering what is this use-case. Could you please explain how this can be useful?

**Comment by tty2** [comment]
:)
Check this [PR](https://github.com/charmbracelet/bubbles/pull/240).
It's one of the way how to implement horizontal scroll.
And the final goal is my personal tui app :) the problem is mentioned in [this issue](https://github.com/charmbracelet/bubbles/issues/236)

**Comment by tty2** [comment]
So instead of using `getStringWithIndent` function that is written by me in the PR above, I'm gonna use this TrimPrefix function that will take care of cell not runes.

**Comment by mattn** [comment]
So `TrimPrefix("あいうえお", 5, "")` must be `" えお"`

```
あいうえお
Aあいうえお
あいうえお
Aあいうえお
```
Result of trimming 5 cells should be
```
 えお
うえお
 えお
うえお
```

**Comment by tty2** [comment]
> So `TrimPrefix("あいうえお", 5, "")` must be `" えお"`
> 
> ```
> あいうえお
> Aあいうえお
> あいうえお
> Aあいうえお
> ```
> 
> Result of trimming 5 cells should be
> 
> ```
>  えお
> うえお
>  えお
> うえお
> ```

Nope.  `TrimPrefix("あいうえお", 5, "")` returns `"うえお"`

And the result of `TrimPrefix` for this:
```
あいうえお
Aあいうえお
あいうえお
Aあいうえお
```
will be the same. I mean `"うえお"`. Because. Prefix width is not bigger than 5.

```
あい:	6 bytes, 2 runes, 4 cells, 2 grapheme clusters
Aあい:	7 bytes, 3 runes, 5 cells, 3 grapheme clusters
あい:	6 bytes, 2 runes, 4 cells, 2 grapheme clusters
Aあい:	7 bytes, 3 runes, 5 cells, 3 grapheme clusters
```

I'd love to find the way to measure exactly 5 cells, but I'm afraid, there is no way to do it.

**Comment by tty2** [comment]
Or...
You probably wanted to say that there is a way how to improve the logic? Are you?
Add one empty cell if there is a difference.

**Comment by tty2** [comment]
I've pushed the implementation with logic from above comment.
And I decided to change the function name to `TruncateLeft`. It's more obvious, especially because there is a `Truncate` function.
Another reason is that there is a function in standard library `strings.TrimPrefix` that cut string relying on `prefix` (if it exists). For users this name similarity can play bad game.
WDYT?

**Comment by mattn** [comment]
Look good to me.

**Comment by mattn** [comment]
Thank you

**Comment by tty2** [comment]
Thank you too

**Pull request merged**